core/storage.go

```diff
@@ -11,41 +11,46 @@ var (
 	ErrNotImplemented = errors.New("method not-implemented")
 )
 
-// ObjectStorage generic storage of objects
-type ObjectStorage interface {
+// ObjectStorer generic storage of objects
+type ObjectStorer interface {
 	// NewObject returns a new Object, the real type of the object can be a
 	// custom implementation or the defaul one, MemoryObject
 	NewObject() Object
-	// Set save an object into the storage, the object shuld be create with
-	// the NewObject, method, and file if the type is not supported.
-	Set(Object) (Hash, error)
-	// Get an object by hash with the given ObjectType. Implementors should
-	// return (nil, ErrObjectNotFound) if an object doesn't exist with both the
-	// given hash and object type.
+	// SetObject save an object into the storage, the object shuld be create
+	// with the NewObject, method, and file if the type is not supported.
+	SetObject(Object) (Hash, error)
+	// GetObject an object by hash with the given ObjectType. Implementors
+	// should return (nil, ErrObjectNotFound) if an object doesn't exist with
+	// both the given hash and object type.
 	//
 	// Valid ObjectType values are CommitObject, BlobObject, TagObject,
-	// TreeObject and AnyObject.
-	//
-	// If AnyObject is given, the object must be looked up regardless of its type.
-	Get(ObjectType, Hash) (Object, error)
-	// Iter returns a custom ObjectIter over all the object on the storage.
+	// TreeObject and AnyObject. If AnyObject is given, the object must be
+	// looked up regardless of its type.
+	GetObject(ObjectType, Hash) (Object, error)
+	// IterObjects returns a custom ObjectIter over all the object on the
+	// storage.
 	//
 	// Valid ObjectType values are CommitObject, BlobObject, TagObject,
-	Iter(ObjectType) (ObjectIter, error)
+	IterObjects(ObjectType) (ObjectIter, error)
+}
+
+// ObjectStorerTx is a optional method for ObjectStorer, it enable transaction
+// base write and read operations in the storage
+type ObjectStorerTx interface {
```

Comment:
We are going too far with the naming of this types, I suggest naming this Tx and move it to an store/object package.